File: Creational/Singleton.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package design.pattern;

/**
 *
 * @author shantanu1395
 */
//Volatile-Thread would not locally cache the value of the variable

class Singleton{
    private static volatile Singleton i;
    private Singleton(){
        System.out.println("Creating instance");
    }
    public static Singleton getInstance(){
        synchronized(Singleton.class){
            if(i==null){
             i=new Singleton();
            }
        }
        return i;
    }
}

class Runner {
    public static void main(String[] args) {
        Singleton s1=Singleton.getInstance();
        System.out.println(s1.hashCode());
        Singleton s2=Singleton.getInstance();
        System.out.println(s2.hashCode());
    }
}

File: Creational/Singleton/singleton.py
from threading import Lock
class Singleton():
    _instance=None
    def __init__(self):
        print("Creating Object..")

    @staticmethod      
    def getInstance():
        _lock = Lock()
        if Singleton._instance==None:
            Singleton._instance=Singleton()
        return Singleton._instance

s1=Singleton.getInstance()    
print(id(s1))

s2=Singleton.getInstance()    
print(id(s2))
